mock/mock.go

```diff
@@ -2678,6 +2686,26 @@ func returnPortGroup(w http.ResponseWriter, portGroupID string) {
 	}
 }
 
+func handleStorageGroupMetrics(w http.ResponseWriter, r *http.Request) {
+	mockCacheMutex.Lock()
+	defer mockCacheMutex.Unlock()
+	if InducedErrors.GetStorageGroupMetricsError {
+		writeError(w, "Error getting storage group metrics: induced error", http.StatusRequestTimeout)
+		return
+	}
+	returnJSONFile(Data.JSONDir, "storage_group_metrics.json", w, nil)
+}
+
+func handleVolumeMetrics(w http.ResponseWriter, r *http.Request) {
+	mockCacheMutex.Lock()
+	defer mockCacheMutex.Unlock()
+	if InducedErrors.GetVolumesMetricsError {
+		writeError(w, "Error getting volume metrics: induced error", http.StatusRequestTimeout)
+		return
+	}
+	returnJSONFile(Data.JSONDir, "volume_metrics.json", w, nil)
```

Comment:
instead of returning the JSON, create an object and use **_writeJSON_** function to return the response.